packages/core/protocols/src/proto/dxos/client/services.proto

```diff
@@ -480,6 +481,9 @@ message Invitation {
   // TODO(nf): some feedback mechanism or GC for immortal invitations?
   // TODO(nf): should the creator have a way to make an invitation last as long as the client? this was the previous behavior.
   optional int32 lifetime = 14;
+
+  /// Whether an invitation can be used multiple times.
+  optional bool multi_use = 15;
```

Comment:
Generally the question that we've asked whenever adding a boolean to the api is what other values might it have later? This is why you'll tend to see more enums in the protos than bools.

Questions I'd have about this are:
- is multiuse meaningfully distinct from timeout? (probably? you might want to have an indefinite timeout but only allow it to be redeemed once, or vice versa)
- is the generic case of this just a integer for the number of uses rather than a boolean?

Reply:
> is the generic case of this just a integer for the number of uses rather than a boolean?

that's a good question and there can probably be legit cases for "n-use invitations" but sounds like an overcomplication since we'll need to also implement "used counter" of a kind to support this

Reply:
actually I'm realizing, what is the use case for multi-use + offline? if it's offline it needs a known public key and if it's got a known public key it can only admit that key so how could it be used multiple times?

Reply:
> that's a good question and there can probably be legit cases for "n-use invitations" but sounds like an overcomplication since we'll need to also implement "used counter" of a kind to support this

even if we don't fully implement the protocol we might consider making it an integer in the protobuf to avoid needing to change the protocol later, while our implementation could omit use-tracking in the mean time

Reply:
known public key is going to belong to an ephemeral keypair. you'll use it to sign a challenge and if signature passes verification your identity will be admitted

Reply:
so different identities can reuse it

Reply:
> known public key is going to belong to an ephemeral keypair

ah I now see why you don't like the term "contact" invitation 😆 

maybe "deferred" or "delegated" would be better than "offline"? I think "offline" doesn't make sense because if you're actually offline/disconnected from peers no one will see the invitation to be able to admit the person you invite even if you are offline when they show up.

Reply:
I like delegated